poh/src/poh_recorder.rs

```diff
@@ -512,30 +506,23 @@ impl PohRecorder {
         false
     }
 
-    fn reached_leader_tick(
-        &self,
-        my_pubkey: &Pubkey,
-        leader_first_tick_height_including_grace_ticks: u64,
-    ) -> bool {
-        if self.start_tick_height + self.grace_ticks
-            == leader_first_tick_height_including_grace_ticks
-        {
+    fn reached_leader_tick(&self, my_pubkey: &Pubkey, leader_first_tick_height: u64) -> bool {
+        if self.start_tick_height == leader_first_tick_height {
             // PoH was reset to run immediately.
             return true;
         }
 
-        let target_tick_height = leader_first_tick_height_including_grace_ticks.saturating_sub(1);
-        if self.tick_height >= target_tick_height {
-            // We have finished waiting for grace ticks.
-            return true;
-        }
-
-        let ideal_target_tick_height = target_tick_height.saturating_sub(self.grace_ticks);
+        let ideal_target_tick_height = leader_first_tick_height.saturating_sub(1);
         if self.tick_height < ideal_target_tick_height {
```

Comment:
as long as we're re-organizing things in here... what do you think of using a match here for the 3 possible tick ranges we consider?

something like:
```
match self.tick_height {
    current_tick if current_tick < ideal_target_tick_height => {
        // We haven't ticked to our leader slot yet.
        false
    }
    current_tick if current_tick >= ideal_target_tick_height.saturating_add(self.grace_ticks) => {
        // We have finished waiting for grace ticks.
        true
    }
    _ => {
        // Let's see if we're allowed to skip grace ticks...
        self.can_skip_grace_ticks(my_pubkey)
    }
}
```

Just something I've been pondering. I'm also fine w/ what you have

Reply:
I prefer what we have already